meeteval/wer/matching/cy_levenshtein.pyx

```diff
@@ -0,0 +1,279 @@
+    if len(reference) != len(reference_timing):
+        raise ValueError(
+            f'reference and reference_timing have mismatching lengths '
+            f'{len(reference)} != {len(reference_timing)}'
+        )
+    if len(hypothesis) != len(hypothesis_timing):
+        raise ValueError(
+            f'hypothesis and hypothesis_timing have mismatching lengths '
+            f'{len(hypothesis)} != {len(hypothesis_timing)}'
+        )
+
+    reference_timing = np.array(reference_timing)
+    hypothesis_timing = np.array(hypothesis_timing)
+
+    assert len(reference) == 0 or reference_timing.shape == (len(reference), 2), (reference_timing.shape, len(reference))
+    assert len(hypothesis) == 0 or hypothesis_timing.shape == (len(hypothesis), 2), (hypothesis_timing.shape, len(hypothesis))
+    assert len(reference) == 0 or len(hypothesis) == 0 or reference_timing.dtype == hypothesis_timing.dtype, (reference_timing.dtype, hypothesis_timing.dtype)
+
+    def check(timing):
+        # end >= start
+        if np.any(timing[:, 1] < timing[:, 0]):
+            raise ValueError(f'The end time of an interval must not be smaller than its begin time')
+        # start values are increasing
+        if np.any(np.diff(timing[:, 0]) < 0):
+            raise ValueError(
+                f'The start times of the annotations must be increasing, which they are not. '
+                f'This might be caused by overlapping segments, see the (potential) previous warning.'
+            )
+
+    if len(reference):
+        check(reference_timing)
+    if len(hypothesis):
+        check(hypothesis_timing)
+        
+    return reference, hypothesis, reference_timing, hypothesis_timing
+
+def time_constrained_levenshtein_distance(
+        reference,  # list[int]
+        hypothesis,  # list[int]
+        reference_timing,  # list[tuple[int, int]]
+        hypothesis_timing,  # list[tuple[int, int]]
+        cost_del=1,
+        cost_ins=1,
+        cost_sub=1,
+        cost_cor=0,
+):
+    reference, hypothesis, reference_timing, hypothesis_timing = _validate_inputs(reference, hypothesis, reference_timing, hypothesis_timing)
+    if len(reference) == 0:
+        return len(hypothesis)
+    if len(hypothesis) == 0:
+        return len(reference)
+    reference, hypothesis = obj2vec(reference, hypothesis)
+
+    args = (reference, hypothesis,
+            reference_timing,
+            hypothesis_timing,
+            cost_del,
+            cost_ins,
+            cost_sub,
+            cost_cor,)
+    if np.issubdtype(reference_timing.dtype, np.signedinteger):
+        return time_constrained_levenshtein_distance_[int](*args)
+    elif np.issubdtype(reference_timing.dtype, np.unsignedinteger):
+        return time_constrained_levenshtein_distance_[uint](*args)
+    elif np.issubdtype(reference_timing.dtype, np.floating):
+        return time_constrained_levenshtein_distance_[float](*args)
```

Comment:
Is this the python float (double precision floating point number with 64 bits) or the c float (single precision floating point number with 32 bits)?

Reply:
We should use double precision

Reply:
I checked, this corresponds to the C++ `float` type. I changed it to `double`